unified-runtime/source/adapters/cuda/enqueue.cpp

```diff
@@ -1104,7 +1104,7 @@ static size_t imageElementByteSize(CUDA_ARRAY_DESCRIPTOR ArrayDesc) {
   case CU_AD_FORMAT_FLOAT:
     return 4;
   default:
-    detail::ur::die("Invalid image format.");
+    die("Invalid image format.");
```

Comment:
I would prefer if we could completely remove all traces of `die()` and use proper error handling mechanisms in their place.

Reply:
Agreed, we also have a weird `detail::ur::assertion` that uses `die()` in the cuda/hip plugin, I want to clean all that up, but it goes a bit beyond the scope of this more "mechanical" PR.

Reply:
makes sense